Libraries/LibWeb/Credentials/CredentialsContainer.cpp

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2024, Miguel Sacristán <miguel_tete17@hotmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "CredentialsContainer.h"
+
+#include <AK/Vector.h>
+#include <LibWeb/Credentials/Credential.h>
+#include <LibWeb/HTML/Window.h>
+#include <LibWeb/Platform/EventLoopPlugin.h>
+
+namespace Web::Credentials {
+GC_DEFINE_ALLOCATOR(CredentialsContainer);
+
+WebIDL::ExceptionOr<GC::Ref<CredentialsContainer>> CredentialsContainer::construct_impl(JS::Realm& realm)
+{
+    return realm.create<CredentialsContainer>(realm);
+}
+
+// https://www.w3.org/TR/credential-management-1/#abstract-opdef-request-a-credential
+GC::Ref<WebIDL::Promise> CredentialsContainer::get(CredentialRequestOptions const& options)
+{
+    // 1. Let settings be the current settings object.
+    auto& settings = HTML::current_principal_settings_object();
+
+    // 2. Assert: settings is a secure context.
+    ASSERT(HTML::is_secure_context(settings));
```

Comment:
For an assert like this from the spec, we want a release-assertion rather than a debug-assertion. So this should be VERIFY, not ASSERT.